src/Simplex/Chat/Library/Commands.hs

```diff
@@ -3400,10 +3457,39 @@ processChatCommand' vr = \case
       CRQueueInfo user msgInfo <$> withAgent (`getConnectionQueueInfo` acId)
     withSendRef :: ChatRef -> (SendRef -> CM ChatResponse) -> CM ChatResponse
     withSendRef chatRef a = case chatRef of
-      ChatRef CTDirect cId -> a $ SRDirect cId
-      ChatRef CTGroup gId -> a $ SRGroup gId Nothing
+      ChatRef CRTDirect cId -> a $ SRDirect cId
+      ChatRef (CRTGroup gcs) gId -> a $ SRGroup gId gcs
       _ -> throwChatError $ CECommandError "not supported"
 
+getGCSDependencies :: VersionRangeChat -> User -> GroupInfo -> GroupChatScope -> VersionChat -> CM (GroupChatScopeInfo, MsgScope, [GroupMember], Int)
```

Comment:
the name is cryptic - what does it do?

Reply:
it seems like it returns a bunch of things that must be derived locally and not in function - e.g., you should never return array and its length - length can be taken locally at a point of use.

Same for MsgScope - it seems like it can be trivially derived from GroupChatScopeInfo locally.

You should not be creating functions to simply do trivial conversions.